pkg/controller/servicemesh/memberroll/controller.go

```diff
@@ -400,9 +404,60 @@ func (r *ReconcileMemberList) Reconcile(request reconcile.Request) (reconcile.Re
 		}
 	}
 
+	// tell Kiali about all the namespaces in the mesh
+	r.reconcileKiali(instance.Namespace, requiredMembers, reqLogger)
+
 	return reconcile.Result{}, err
 }
 
+func (r *ReconcileMemberList) reconcileKiali(kialiCRNamespace string, requiredMembers map[string]struct{}, reqLogger logr.Logger) error {
+
+	reqLogger.Info("Attempting to get Kiali CR", "kialiCRNamespace", kialiCRNamespace)
+
+	kialiCR := &unstructured.Unstructured{}
+	kialiCR.SetAPIVersion("kiali.io/v1alpha1")
+	kialiCR.SetKind("Kiali")
+	kialiCR.SetNamespace(kialiCRNamespace)
+	kialiCR.SetName("kiali")
+	err := r.Client.Get(context.TODO(), client.ObjectKey{Name: "kiali", Namespace: kialiCRNamespace}, kialiCR)
+	if err != nil {
+		if errors.IsNotFound(err) || errors.IsGone(err) {
+			reqLogger.Info("Kiali CR does not exist, Kiali probably not enabled")
+			return nil
+		}
+		reqLogger.Error(err, "error retrieving Kiali CR from mesh")
+		return err
+	}
+
+	// just get an array of strings consisting of the list of namespaces to be accessible to Kiali
+	var accessibleNamespaces []string
+	if len(requiredMembers) == 0 {
+		// we are not in multitenency mode - kiali can access the entire cluster
+		accessibleNamespaces = []string{"**"}
+	} else {
+		// we are in multitenency mode
+		accessibleNamespaces = make([]string, 0, len(requiredMembers))
+		for key := range requiredMembers {
+			accessibleNamespaces = append(accessibleNamespaces, key)
```

Comment:
Would that work when the namespace from requiredMembers?

Reply:
Yes. This is happening in the function "reconcileKiali" - it is getting called during a reconciliation ... so when you change the SMMR by adding or removing from the spec.members list, this gets called with the new list "requiredMembers".

I tested this part and it seems to work pretty well.

Reply:
Hmm... one thing I did not test is if you remove ALL the namespaces from spec.members (when spec.members is empty, I think Maistra will assume only the control plane namespace is in the mesh). However, this should still work since `accessibleNamespaces` will be of type `[]string]` but will be empty - so the line down below that calls `SetNestedStringSlice` should be setting the spec.deployment.accessible_namespaces yaml correctly (i.e. it should be set to `[ ]` - an empty list).